server/methods.js

```diff
@@ -168,12 +175,80 @@ Meteor.methods({
     const regex = new RegExp(/^(?=.*[A-Z0-9])/i);
     if (regex.test(review._id) && userIsAdmin) {
       Reviews.remove({ _id: review._id });
+      Meteor.call("updateCourseMetrics", review.class);
       return 1;
     } else {
       return 0;
     }
   },
 
+  // This updates the metrics for an individual class given its Mongo-generated id. 
+  // Returns 1 if successful, 0 otherwise.
+  updateCourseMetrics: function (courseId) {
+    let course = Meteor.call('getCourseById', courseId);
+    if (course) {
+      let reviews = Reviews.find({ class: courseId, reported:0, visible:1}).fetch();
+      let state = getGaugeValues(reviews);
+      Classes.update({ _id: courseId },
+        {
+          $set: {
+            //If no data is available, getGaugeValues returns "-" for metric
+            classDifficulty: (state.diff !== "-" ? Number(state.diff) : null ),
+            classRating: (state.rating !== "-" ? Number(state.rating) : null ),
+            classWorkload: (state.workload !== "-" ? Number(state.workload) : null )
+          }
+        });
+      return 1;
+
+    }
+    else {
+      return 0;
+    }
+
+  },
```

Comment:
This endpoint is unprotected. Anyone can call it.

Reply:
@draycfarley can you add a "token" param and admin validation condition like we use to protect the other endpoints?

Reply:
Fixed by @draycfarley  more recent commit.